pkg/sdkserver/sdkserver.go

```diff
@@ -491,14 +492,36 @@ func (s *SDKServer) GetGameServer(context.Context, *sdk.Empty) (*sdk.GameServer,
 		return nil, err
 	}
 
-	return convert(gs), nil
+	s.cachedGameServer = convert(gs)
+	return s.cachedGameServer, nil
+}
+
+func (s *SDKServer) getCachedGameServer() (*sdk.GameServer, error) {
+	if s.cachedGameServer != nil {
+		return s.cachedGameServer, nil
+	}
+
+	return s.GetGameServer(context.Background(), &sdk.Empty{})
 }
 
 // WatchGameServer sends events through the stream when changes occur to the
 // backing GameServer configuration / status
 func (s *SDKServer) WatchGameServer(_ *sdk.Empty, stream sdk.SDK_WatchGameServerServer) error {
 	s.logger.Debug("Received WatchGameServer request, adding stream to connectedStreams")
 	s.streamMutex.Lock()
+
+	if runtime.FeatureEnabled(runtime.FeatureGameServerCaching) {
+		gs, err := s.getCachedGameServer()
+		if err != nil {
+			s.logger.WithError(errors.WithStack(err)).Error("error getting cached game server")
```

Comment:
I think it would be better to return the error here instead. (Maybe wrap it, maybe not?)

Seems like it shouldn't be swallowed like this. If it fails, there's a good reason.